docs/src/components/FlutterAuthenticatorExample.tsx

```diff
@@ -12,15 +14,20 @@ export function FlutterAuthenticatorExample({
   // it is used by the authenticator to signal when the authenticator has finished loading.
   id = generateId(),
 }) {
-  const colorAttr = 'data-amplify-color-mode';
-  // in dev mode, `data-amplify-color-mode` will not be on the html element
-  const colorElements =
-    typeof window !== 'undefined'
-      ? document.querySelectorAll(`[${colorAttr}]`)
-      : [];
-  const themeMode = colorElements.length
-    ? colorElements[0].getAttribute(colorAttr)
-    : 'light';
+  const [themeMode, setThemeMode] = React.useState('light');
+  React.useLayoutEffect(() => {
+    let themeMode = document.documentElement.getAttribute(
+      'data-amplify-color-mode'
+    );
+    if (themeMode === 'system') {
+      if (window.matchMedia('(prefers-color-scheme:dark)').matches) {
+        themeMode = 'dark';
+      } else {
+        themeMode = 'light';
+      }
+    }
+    setThemeMode(themeMode);
+  }, []);
```

Comment:
Without full context might be missing something but it looks like `themeMode` could be resolved with a ternary, e.g:
```ts
 const themeMode = window.matchMedia('(prefers-color-scheme:dark)').matches) ? 'dark' : 'light';
```